### PR TITLE
Rename ambiguous "atom" argument to "tag" in Record module functions

### PR DESCRIPTION
**Pull request by eksperimental**
"atom" could be either the "name" or "tag" of a record.

### CODE REVIEW

File: lib/elixir/lib/record.ex [hunk 1]
```diff
@@ -312,50 +312,50 @@ defmodule Record do
...
     cond do
       is_atom(args) ->
-        get(atom, fields, record, args)
+        get(tag, fields, record, args)
       Keyword.keyword?(args) ->
-        update(atom, fields, record, args, caller)
+        update(tag, fields, record, args, caller)
       true ->
-        msg = "expected arguments to be a compile time atom or keywords, got: #{Macro.to_string args}"
+        msg = "expected arguments to be a compile time atom or a keyword list, got: #{Macro.to_string args}"
```
Comment (eksperimental) [inline lib/elixir/lib/record.ex]: please see if this message suits better

**Comment by josevalim** [comment]
It has to be an atom: https://github.com/eksperimental/elixir/blob/cd7ce792b36f660f90f9fb189e9cbcfd63c8cd8b/lib/elixir/lib/record.ex#L161

**Comment by eksperimental** [comment]
@josevalim I don't quiet understand what you are referring to

**Comment by josevalim** [comment]
The is_record implementation is going to check if the first element is an atom, otherwise it returns false.

**Comment by fishcakez** [comment]
I think @eksperimental is renaming the variable to refer to the tagged tuple format of a record. Much like one might rename a variable `list` to a more descriptive name like `children`.

**Comment by eksperimental** [comment]
precisely @fishcakez!
thank you for the clarification

**Comment by josevalim** [comment]
I see. Thank you both!

**Comment by whatyouhide** [comment]
Thanks @eksperimental! 💟

**Pull request merged**